homeassistant/components/sma/__init__.py

```diff
@@ -1 +1,193 @@
-"""The sma component."""
+"""The sma integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import pysma
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PATH,
+    CONF_SCAN_INTERVAL,
+    CONF_SENSORS,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_CUSTOM,
+    CONF_FACTOR,
+    CONF_GROUP,
+    CONF_KEY,
+    CONF_UNIT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    PLATFORMS,
+    PYSMA_COORDINATOR,
+    PYSMA_OBJECT,
+    PYSMA_REMOVE_LISTENER,
+    PYSMA_SENSORS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _parse_legacy_options(entry: ConfigEntry, sensor_def: pysma.Sensors) -> None:
+    """Parse legacy configuration options.
+
+    This will parse the legacy CONF_SENSORS and CONF_CUSTOM configuration options
+    to support deprecated yaml config from platform setup.
+    """
+
+    # Add sensors from the custom config
+    sensor_def.add(
+        [
+            pysma.Sensor(o[CONF_KEY], n, o[CONF_UNIT], o[CONF_FACTOR], o.get(CONF_PATH))
+            for n, o in entry.data.get(CONF_CUSTOM).items()
+        ]
+    )
+
+    # Parsing of sensors configuration
+    config_sensors = entry.data.get(CONF_SENSORS)
+    if config_sensors:
```

Comment:
```suggestion
    if not config_sensors:
        return
```

Then outdent the block below

Reply:
Fixed:
https://github.com/home-assistant/core/commit/2ad7ad8c9aae7286e00dd578cbf2342eac6bcbc4#diff-a7bdb8a0a15589eb3d6f0a9b41a9131ef4a2563166a67a1536b973a7534adcfeL59-R71